substratevm/src/com.oracle.svm.reflect/src/com/oracle/svm/reflect/serialize/SerializationSupport.java

```diff
@@ -0,0 +1,222 @@
+        // Checkstyle: stop
+        if (exceptionsMsg.length() > 0) {
+            String option = SubstrateOptionsParser.commandArgument(NativeImageOptions.ReportUnsupportedElementsAtRuntime, "+");
+            if (!NativeImageOptions.ReportUnsupportedElementsAtRuntime.getValue()) {
+                BigBang bb = access.getBigBang();
+                bb.getUnsupportedFeatures().addMessage(MULTIPLE_CHECKSUMS, null,
+                                exceptionsMsg + "\n" + "To allow continuing compilation with above unsupported features, set " + option);
+            } else {
+                System.out.println(exceptionsMsg);
+                System.out.println("Compilation will continue because " + option +
+                                " was set. But the program may behave unexpectedly at runtime.");
+            }
+        }
+        // Checkstyle: resume
+    }
+
+    @Platforms({Platform.HOSTED_ONLY.class})
+    @Override
+    public Class<?> addSerializationConstructorAccessorClass(Class<?> serializationTargetClass, Long configuredChecksum) {
+        if (serializationTargetClass.isArray() || serializationTargetClass.isEnum()) {
+            return null;
+        }
+        // Don't generate SerializationConstructorAccessor class for Externalizable case
+        if (Externalizable.class.isAssignableFrom(serializationTargetClass)) {
+            try {
+                Method getExternalizableConstructor = ObjectStreamClass.class.getDeclaredMethod("getExternalizableConstructor", Class.class);
+                getExternalizableConstructor.setAccessible(true);
+                Constructor<?> c = (Constructor<?>) getExternalizableConstructor.invoke(null, serializationTargetClass);
+                return c.getDeclaringClass();
+            } catch (Exception e) {
+                VMError.shouldNotReachHere();
+            }
+        }
+
+        String targetClassName = serializationTargetClass.getName();
+        SerializationKey serializationKey = new SerializationKey(targetClassName, configuredChecksum);
+        ReflectionFactory reflFactory = ReflectionFactory.getReflectionFactory();
+        Constructor<?> buildTimeConstructor = reflFactory.newConstructorForSerialization(serializationTargetClass);
+        Class<?> buildTimeConsClass = buildTimeConstructor.getDeclaringClass();
+        if (configuredChecksum != null) {
+            long buildTimeChecksum = getSerializationChecksum(buildTimeConsClass.getName());
+            if (configuredChecksum.longValue() != buildTimeChecksum) {
+                StringBuilder sb = new StringBuilder();
+                sb.append(targetClassName).append(": configured checksum is ").append(configuredChecksum).append(", build time checksum is ").append(buildTimeChecksum);
+                verifyChecksumErrors.add(sb.toString());
+            }
+        } else {
+            verifyChecksumErrors.add("Checksum for class " + targetClassName + " was not set in configuration");
```

Comment:
As I mentioned elsewhere, I don't think the checksum should be required (although the agent should set it).

Reply:
Replied at https://github.com/oracle/graal/pull/2730#discussion_r489886735